#!/usr/bin/env python
PACKAGE = "mjpeg_cam"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("autoexposure", bool_t, 0, "Turn on/off auto exposure", False)
gen.add("exposure", int_t, 0, "Exposure", 100, 0, 255)
gen.add("brightness", int_t, 0, "Brightness", 128, -64, 64)
gen.add("gamma", int_t, 0, "Gamma", 100, 72, 500)
gen.add("gain", int_t, 0, "gain", 0, 0, 100)
gen.add("saturation", int_t, 0, "Saturation", 56, 0, 128)
gen.add("contrast", int_t, 0, "Contrast", 32, 0, 64)
gen.add("auto_white_balance", bool_t, 0, "AWB", False)
gen.add("white_balance", int_t, 0, "White Balance", 4600, 2800, 6500)

exit(gen.generate(PACKAGE, "mjpeg_cam", "mjpeg_cam"))
